### ISSUE TITLE
flake8 ignores config files

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#780]
In GitLab by @arielpontes on Nov 27, 2020, 24:59

# Steps

I installed `flake8`:

```
pip install flake8
```

Created a `tox.ini` file:

```
[flake8]
exclude = */migrations/*
```

Then I ran `flake8`:

```
flake8
```

# Expected results

Only get errors outside migration directories.

# Actual results

I still get errors inside migration directories. When I run the command manually, it works:

```
flake8 --exclude=*/migrations/*
```

I get other errors, but not inside migration folders.

# Bug report output

```
{
  "dependencies": [],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.6.10",
    "system": "Linux"
  },
  "plugins": [
    {
      "is_local": false,
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "is_local": false,
      "plugin": "pycodestyle",
      "version": "2.6.0"
    },
    {
      "is_local": false,
      "plugin": "pyflakes",
      "version": "2.2.0"
    }
  ],
  "version": "3.8.4"
}
```

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Nov 27, 2020, 06:20

On your linux shell running `--exclude=*/migrations/*` will cause your shell to expand that for us. I'd try a more equivalent `--exclude='*/migrations/*'`. I believe the pattern your using doesn't work well with the standard library we're using to match these. You would probably be better served trying `--exclude=migraitons/*`.

Further, if you ran `flake8 -vv` you'd likely see we are detecting and parsing your config file.

**Comment by asottile** [comment]
In GitLab by @arielpontes on Nov 27, 2020, 07:12

I tried `--exclude=migrations/*` and it didn't work. Not even passing it manually as an explicit parameter to `flake8`. Also, I don't think it's a pattern problem because I've also tried other patterns like `*.py` and the results are the same. The direct argument works, but adding it to the file has no effect.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Nov 27, 2020, 11:01

Did you bother trying to debug this like I said you could with `flake8 -vv`? You're insisting on a bug that with the very minimal information you've provided doesn't make sense.

**Comment by asottile** [comment]
In GitLab by @asottile on Nov 27, 2020, 11:03

fwiw I commented on the SO post -- I cannot reproduce given what you've showed: https://stackoverflow.com/questions/65035018/flake8-ignoring-configuration-files#comment114985868_65035018

**Comment by asottile** [comment]
In GitLab by @arielpontes on Nov 28, 2020, 02:08

Yes Ian, I did bother to debug but the output was so much that I didn't find it helpful at first. I just ran it again now though and spent some more time going through it and I found this line:

```
Local configuration files have no flake8 section
```

This led me to dig deeper and it turns out it was a stupid mistake: I had messed up my docker-compose file and wasn't sharing the volumes properly, so the new `tox.ini` file wasn't available in the container 😅 Sorry for wasting your time on this one. Thanks for the debugging tips though, it helped me identify the problem faster.
